### PR TITLE
Added Record protocol, updated CloudKitStore with methods, added basic UserController

### PR DESCRIPTION
**Pull request by wfltaylor**
Note that the UserRecord uses custom fields that you have to manually add on the CloudKit dashboard.

### CODE REVIEW

File: Project SF/Models/CloudKit/CloudKitStore.swift [hunk 1]
```diff
@@ -90,6 +112,33 @@ class CloudKitStore {
         }
     }
     
+    /// Asynchronously fetches the user record from the CloudKit database
+    /// - Parameter handler: Called with the result of the operation. Not guaranteed to be on the main thread.
+    func fetchUserRecord(then handler: @escaping (Result<UserRecord, Error>) -> Void) {
```
Comment (RomanEsin) [inline Project SF/Models/CloudKit/CloudKitStore.swift]: Why is it using an `Error` in the result instead of `CloudKitStoreError `?

**Comment by RomanEsin** [comment]
#69 nice

**Comment by RomanEsin** [comment]
Maybe do something like this in commit? @wfltaylor

**Comment by wfltaylor** [comment]
Okay, I'll work on it tomorrow

**Comment by RomanEsin** [comment]
Good ugh!

**Comment by RomanEsin** [comment]
Goodnight!

**Review by RomanEsin — APPROVED** [review]

**Review by RomanEsin — APPROVED** [review]

File: Project SF/Project SF.entitlements [hunk 2]
```diff
@@ -10,7 +10,7 @@
 	<array/>
 	<key>com.apple.developer.icloud-container-identifiers</key>
 	<array>
-		<string>iCloud.ProjectSF</string>
+		<string>iCloud.com.wfltaylor.projectsf</string>
```
Comment (RomanEsin) [inline Project SF/Project SF.entitlements]: May we keep this `iCloud.ProjectSF` on everyone's commits so we don't need to change every time?

File: Project SF/Views/Tabs/Settings/Profile Settings/ProfileSettings.swift [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    @StateObject var controller = ProfileSettingsController()
+    
+    // MARK: View
+
+    var body: some View {
+        ScrollView {
+            switch controller.userController.state {
+            case .loading:
+                ProgressView()
+            case .user(_):
```
Comment (RomanEsin) [inline Project SF/Views/Tabs/Settings/Profile Settings/ProfileSettings.swift]: Could you also resolve a SwiftLint ⚠️  here?

**Comment by Priva28** [comment]
Your code looks all good!

**Review by RomanEsin — APPROVED** [review]

**Comment by Priva28** [comment]
I think we should consider merging what has been done so far into development because there are gonna be heaps of conflicts that is better to resolve now I guess.

**Review by Priva28 — APPROVED** [review]

**Comment by Priva28** [comment]
Wow that was a surprisingly painless merge 🎉

Maybe still create a new PR for more CK changes and continue committing to the CloudKit branch.

**Pull request merged**